adapters/verizonmedia/verizonmedia.go

```diff
@@ -0,0 +1,189 @@
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+)
+
+type VerizonMediaAdapter struct {
+	http *adapters.HTTPAdapter
+	URI  string
+}
+
+func (a *VerizonMediaAdapter) Name() string {
+	return "verizonmedia"
+}
+
+func (a *VerizonMediaAdapter) SkipNoCookies() bool {
+	return false
+}
+
+func (a *VerizonMediaAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+
+	errs := make([]error, 0, len(request.Imp))
+	if len(request.Imp) == 0 {
+		err := &errortypes.BadInput{
+			Message: "No impression in the bid request",
+		}
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	errors := make([]error, 0, 1)
+
+	var bidderExt adapters.ExtImpBidder
+	err := json.Unmarshal(request.Imp[0].Ext, &bidderExt)
+	if err != nil {
+		err = &errortypes.BadInput{
+			Message: "ext.bidder not provided",
+		}
+		errors = append(errors, err)
+		return nil, errors
+	}
+	var verizonMediaExt openrtb_ext.ExtImpVerizonMedia
+	err = json.Unmarshal(bidderExt.Bidder, &verizonMediaExt)
+	if err != nil {
+		err = &errortypes.BadInput{
+			Message: err.Error(),
+		}
+		errors = append(errors, err)
+		return nil, errors
+	}
+
+	if verizonMediaExt.Dcn == "" {
+		err = &errortypes.BadInput{
+			Message: "Missing param dcn",
+		}
+		errors = append(errors, err)
+		return nil, errors
+	}
+
+	if verizonMediaExt.Pos == "" {
+		err = &errortypes.BadInput{
+			Message: "Missing param pos",
+		}
+		errors = append(errors, err)
+		return nil, errors
+	}
+
+	siteCopy := *request.Site
+	request.Site = &siteCopy
+	changeRequestForBidService(request, &verizonMediaExt)
+	reqJSON, err := json.Marshal(request)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	thisURI := a.URI
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Set("User-Agent", request.Device.UA)
+	headers.Add("x-openrtb-version", "2.5")
+
+	return []*adapters.RequestData{{
+		Method:  "POST",
+		Uri:     thisURI,
+		Body:    reqJSON,
+		Headers: headers,
+	}}, errors
+}
+
+func (a *VerizonMediaAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. ", response.StatusCode),
+		}}
+	}
+
+	var bidResp openrtb.BidResponse
+	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Bad server response: %d. ", err),
+		}}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(1)
+
+	for _, sb := range bidResp.SeatBid {
+		for _, bid := range sb.Bid {
+			if !impIDExists(bid.ImpID, internalRequest.Imp) {
```

Comment:
Do we need a bigger initial size for our bids array? After all, if we expect to have more than one Bid most of the times maybe we want to allocate a buffer that makes sense in our expected bid number context so we don't have to dynamically allocate elements on the heap via the `append` function.
```
 99 func (a *VerizonMediaAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
100 +-- 17 lines: if response.StatusCode == http.StatusNoContent {----------------------------------
117
118     bidResponse := adapters.NewBidderResponseWithBidsCapacity(1)
                       --------------------------------------
                        63 func NewBidderResponseWithBidsCapacity(bidsCapacity int) *BidderResponse {
                        64     return &BidderResponse{
                        65         Currency: "USD",
                        66         Bids:     make([]*TypedBid, 0, bidsCapacity),
                        67     }
                        68 }
                       adapters/bidder.go
                       --------------------------------------
119
120     for _, sb := range bidResp.SeatBid {
121         for _, bid := range sb.Bid 
122             if !impIDExists(bid.ImpID, internalRequest.Imp) {
adapters/verizonmedia/verizonmedia.go
```

Reply:
Going with the number of Imps in the request may not be a bad idea.

Reply:
I used the len of the request.